src/main/java/uk/ac/ebi/ega/egafuse/config/EgaFuseApplicationConfig.java

```diff
@@ -46,13 +44,22 @@
 @Configuration
 @EnableCaching
 public class EgaFuseApplicationConfig {
+    
+    public static final long PAGE_SIZE = 1024L * 1024L * 10L;
+    public static int NUM_PAGES;
+
+    @Value("${connection}")
+    public void setNameStatic(int connection){
+        NUM_PAGES = connection;

```

Comment:
Do not tie the number of pages to the connections. The limit of the connections is on the pool of connections of your http client. The number of pages should be to limit the ammount of memory that the application can use as a cache. In this case basically the default configuration would only have 4 pages!!! of 10MiB so a cache of 40MiB for the kind of data we have is virtually useless.

Reply:
It is not like that. 
@Value("${maxCache}") is the total cache size which I'm initializing in CacheManager()

Here we are getting connection which I use as a number of pages parallelly download in the background as shown here `https://github.com/EGA-archive/ega-fuse/blob/amohan/EE-1134/src/main/java/uk/ac/ebi/ega/egafuse/service/EgaFile.java#L86`

So if maxCache = 100 & connection =4 
It means that 10 MB * maxCache  = 1000 MB total cache size 
and simultaneously 4 pages are downloaded and stored in the cache.

Reply:
Then first why you call it NUM_PAGES? it does not relate at all.
Second why you have that second configuration for the limit of threads when your pool of connection will already limit your ability to get the data?
I see that you are using a thread manager to execute and fill it up. Worse, there is no blocking  to wait for the data to arrive in your current thread...

The way it is written it will only work if you have a very early response. I suggest you to take a look at how to use an AsyncCache -> https://github.com/ben-manes/caffeine/wiki/Population#asynchronously-loading

Reply:
Done